### PR TITLE
Update inventory ssd

### PR DESCRIPTION
**Pull request by jstac**
Misc improvements -- make it nicer

Finish py file and then convert back to myst with jupytext

Clean up the vectorization stuff and different versions of B.

Get the plot working again and add some brief comments at the end.

### CODE REVIEW

**Comment by HumphreyYang** [comment]
Hi @jstac,

It looks great to me! For compilation and execution time, I got 

```
compile plus execution time = 3930.385113 ms
```

```
execution time = 1590.973854 ms
```

on Tesla T4, which are reasonable to me.

I just pushed some minor cosmetic changes. I applied `block_until_ready` on both  `v_star` and `σ_star`:


```
v_star, σ_star = solve_inventory_model(v_init, model)

# Pause until execution finishes
jax.tree_util.tree_map(lambda x: x.block_until_ready(), (v_star, σ_star))
```

**Comment by HumphreyYang** [comment]
Hi @jstac,

I just had another look, and I think I can optmize this slightly more. I will work on it and report back!

**Comment by mmcky** [comment]
@HumphreyYang I have noticed the preview is referencing `python=3.11` for some reason I can't yet explain. I will try and regenerate the cache.

**Comment by jstac** [comment]
> Hi @jstac,
> 
> I just had another look, and I think I can optmize this slightly more. I will work on it and report back!

Looking forward to seeing the new version :-)

**Comment by jstac** [comment]
Thanks @HumphreyYang, this is very cool.  

Daisuke Oyama used the same strategy for the DP code in quantecon.py -- precompting reward and transition matrices to reduce execution time.  As you show, the speed gain can be really impressive.

Nonetheless, I think we should resist this approach here, since, as you point out, memory issues can be a problem (some people ran into this limitation when using the QE library) and the code is a bit further away from the maths.

(The vmap code is also messy in some ways, but the right hand side of the bellman equation is still easy to understand.)

(Let's also resist adding methods to our Model class in all QE lectures, just to keep everything as consistent as possible.  Once we start adding methods, it's not clear where to stop.)

One small issue that needs to be fixed: we use z as the discount factor without explanation.  Somewhere we need to add that we use beta(z) = z.

**Comment by jstac** [comment]
Thanks @HumphreyYang, I appreciate it.
